src/clib/lib/config/config_parser.cpp

```diff
@@ -565,31 +574,54 @@ config_parse__(config_parser_type *config, config_content_type *content,
 
             // Add define
             else if (define_kw && (strcmp(define_kw, kw) == 0)) {
-                if (active_tokens < 3)
-                    util_abort("%s: keyword:%s must have exactly one (or more) "
-                               "arguments. \n",
-                               __func__, define_kw);
-
-                char *key = (char *)util_alloc_string_copy(
-                    stringlist_iget(token_list, 1));
-                char *value = stringlist_alloc_joined_substring(
-                    token_list, 2, active_tokens, " ");
+                if (active_tokens < 3) {
+                    content->parse_errors.push_back(fmt::format(
+                        "Keyword:{} must have two or more arguments.",
+                        define_kw));
+                    return;
+                }
 
-                config_content_add_define(content, key, value);
+                const std::string key = util_alloc_string_copy(
```

Comment:
Is it necessary to util_allow_string_copy this? Also, should you free the char* string if so?

Reply:
Ah, no, not any more as assigning `char*` to `std::string` is a copy action. Fixed it :)